src/transformStateWithClones.js

```diff
@@ -7,7 +7,24 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let obj = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      obj = Object.assign(obj, action.extraData);
+    } else if (action.type === 'removeProperties') {
+      for (const char of action.keysToRemove) {
+        delete obj[char];
+      }
+    } else if (action.type === 'clear') {
+      obj = {};
+    }
```

Comment:
Use [`switch case`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch) in cases where you constantly compare one value with others